targets/simple_switch/simple_switch.cpp

```diff
@@ -188,6 +188,7 @@ SimpleSwitch::start_and_return_() {
     threads_.push_back(std::thread(&SimpleSwitch::egress_thread, this, i));
   }
   threads_.push_back(std::thread(&SimpleSwitch::transmit_thread, this));
+  threads_.push_back(std::thread(&SimpleSwitch::periodic_thread, this));
```

Comment:
I'm not sure why we need any change in simple_switch at all. Shouldn't the periodic thread be managed entirely by the `SwitchWContexts` base class?

Reply:
Documentation for in `SwitchWContexts` states that the `start_and_return()` function of an individual target should start all processing threads. Is it acceptable to do processing in `SwitchWContexts` as well?

Reply:
There are many threads that are already managed by `SwitchWContexts`. All targets can benefit from this kind of periodic task registration and you can hide thread management from the target (it's already done for learning for example).